src/core/SideNavigation/SidenavButton.tsx

```diff
@@ -23,43 +23,47 @@ export type SidenavButtonProps = {
   isSubmenuOpen?: boolean;
 } & Omit<ButtonProps, 'styleType' | 'size'>;
 
+type SideNavButtonComponent = PolymorphicForwardRefComponent<
+  'button',
+  SidenavButtonProps
+>;
+
 /**
  * Wrapper around Button to be used as SideNavigation items.
  * Label is hidden when sidenav is collapsed.
  */
-export const SidenavButton = React.forwardRef<
-  HTMLButtonElement,
-  SidenavButtonProps
->((props, ref) => {
-  const {
-    className,
-    children,
-    isActive = false,
-    disabled = false,
-    isSubmenuOpen = false,
-    ...rest
-  } = props;
+export const SidenavButton: SideNavButtonComponent = React.forwardRef(
```

Comment:
Tests for this one?

Reply:
Hmm, maybe I need to create a `SidenavButton.test.tsx`. 

But it just delegates the `as` prop to `Button` so maybe it's fine